jwst/outlier_detection/utils.py

```diff
@@ -20,177 +26,365 @@
+            section_nrows = 1
+        self.buffer_size = buffer_size
+
+        nsections = int(np.ceil(imrows / section_nrows))
+        log.info(f"Computing median over {self._expected_nframes} groups in {nsections} "
+                    f"sections with total memory buffer {buffer_size / _ONE_MB} MB")
+        return nsections, section_nrows
+
+
+    def _temparray_setup(self, dtype):
+        """Set up temp file handlers for each spatial section"""
+        temp_arrays = []
+        for i in range(0, self.nsections):
+            shp = self.slice_shape
+            if i == self.nsections - 1:
+                # last section has whatever shape is left over
+                shp = (self.frame_shape[0] - (self.nsections-1) * self.section_nrows, self.frame_shape[1])
+            arr = DiskAppendableArray(shp, dtype, self._temp_path / f"{i}.bin")
+            temp_arrays.append(arr)
+        return temp_arrays
+
+
+    def add_image(self, data):
+        """Split resampled model data into spatial sections and write to disk."""
+        if self._n_adds >= self.nsections:
+            raise IndexError(f"Too many calls to add_image. Expected at most {self.nsections} input models.")
+        self._validate_data(data)
+        self._n_adds += 1
+        for i in range(self.nsections):
+            row1 = i * self.section_nrows
+            row2 = min(row1 + self.section_nrows, self.frame_shape[0])
+            arr = self._temp_arrays[i]
+            arr.append(data[row1:row2])
+
+
+    def _validate_data(self, data):
+        if data.shape != self.frame_shape:
+            raise ValueError(f"Data shape {data.shape} does not match expected shape {self.frame_shape}")
+        if data.dtype != self.dtype:
+            raise ValueError(f"Data dtype {data.dtype} does not match expected dtype {self.dtype}")
+        
 
-    section_nrows : int
-        The number of rows in each section
+    def cleanup(self):
+        """Remove the temporary files and directory when finished"""
+        #[arr.cleanup() for arr in self._temp_arrays]
+        self._temp_dir.cleanup()
+        return
 
-    imrows : int
-        The total number of rows in the image
 
-    Yields
-    ------
-    data_subset : ndarray
-        array of shape (len(library), section_nrows, ncols) representing a spatial
-        subset of all the data arrays in the library
+    def compute_median(self):
+        """Read spatial sections from disk and compute the median across groups
+        (median over number of exposures on a per-pixel basis)"""
+        row_indices = [(i * self.section_nrows, min((i+1) * self.section_nrows, self.frame_shape[0]))
+                       for i in range(self.nsections)]
 
-    weight_subset : ndarray
-        weights corresponding to data_list
+        output_rows = row_indices[-1][1]
+        output_cols = self._temp_arrays[0].shape[2]
+        median_image = np.empty((output_rows, output_cols), self.dtype) * np.nan
```

Comment:
This multiplication might throw InvalidValue warnings sometimes - better to make the array full of NaNs from the start.
```suggestion
        median_image = np.full((output_rows, output_cols), np.nan, dtype=self.dtype)
```